### PR TITLE
Add fully-functional site comments using Datastore

### LINKED ISSUE #14: Add comments to site

### PR DESCRIPTION
**Pull request by mirrorkeydev**
Closes #14.

Adds fully functional comments to the website (Contact page). Comments are stored using Datastore, and persist across deployments.

Simple server-side validation ensures that no empty comments are submitted, and comments without authors are submitted as 'Anonymous'.

Screenshot 2020-06-29 at 4 09 28 PM

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,18 +34,77 @@
...
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String body = getParameter(request, "comment", "");
+        String author = getParameter(request, "name", "Anonymous");
+        Date datetime = new Date();
+
+        // Check for validity
+        if (!body.isEmpty()){
+
```
Comment (mrboxtobox) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You can remove the space here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -27,18 +34,77 @@
...
+    /** POST a new comment to the server */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String body = getParameter(request, "comment", "");
+        String author = getParameter(request, "name", "Anonymous");
+        Date datetime = new Date();
+
+        // Check for validity
```
Comment (mrboxtobox) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Super nit. But end comments with periods.

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -16,15 +16,43 @@ a {
...
   text-align: center;
+  width: 100%;
+}
+
+input,
+textarea {
+  background-color: #f7f7f7;
+  border: 0px;
+  color: gray;
+  border-radius: 5px;
```
Comment (mrboxtobox) [inline portfolio/src/main/webapp/style.css:34]: Nit. It's recommended to order the properties alphabetically.

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -16,15 +16,43 @@ a {
...
+  margin: 5px 0px 5px 0px;
```
Comment (mrboxtobox) [inline portfolio/src/main/webapp/style.css]: This would be simpler: `margin: 5px 0px`

**Review by mrboxtobox — APPROVED** [review]
Very nice!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (mirrorkeydev) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Fixed!

File: portfolio/src/main/webapp/style.css (hunk 3 above)
Comment (mirrorkeydev) [reply portfolio/src/main/webapp/style.css:34]: Fixed!

File: portfolio/src/main/webapp/style.css (hunk 4 above)
Comment (mirrorkeydev) [reply portfolio/src/main/webapp/style.css]: Fixed!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (mirrorkeydev) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Fixed!

**Pull request merged**